mllib/src/main/scala/org/apache/spark/ml/feature/CountVectorizer.scala

```diff
@@ -160,6 +187,11 @@ class CountVectorizer @Since("1.5.0") (@Since("1.5.0") override val uid: String)
     } else {
       $(minDF) * input.cache().count()
     }
+    val maxDf = if ($(maxDF) >= 1.0) {
+      $(maxDF)
+    } else {
+      $(maxDF) * input.cache().count()
+    }
```

Comment:
I wonder if it's best to check that the max is >= the min? this may be the only place that can be done as one might be a fraction and the other not.

Also this ends up counting the input twice when both params are set, which might be worth avoiding. It could be computed and saved only when needed.

(I suppose this also fails to unpersist `input`, which could happen right after `wordCounts.count()`. That's a slightly different issue, so not too concerned about whether to fix it here or not.)

Reply:
Good points.
- I added a check that maxDF >= minDF
- Changed the code so that counting (and caching) is done only once
- I refactored the code so that "filter()" is only invoked if minDF or maxDF is set
- Added an un-persisting the input after the counting is done.